scripts/docs_config/introduction.rst

```diff
@@ -0,0 +1,101 @@
+==============
+ Introduction
+==============
+
+Motivation
+==========
+
+The amount of data associated with modern workloads that need to be processed 
+by modern workloads is continuously growing. To address the increasing demand 
+memory subsystem of modern server platforms is becoming heterogeneous. For 
+example, High-Bandwidth Memory (HBM) introduced in Sapphire Rapids addresses 
+throughput needs; the emerging CXL protocol closes the capacity gap and tends 
+to better memory utilization by memory pooling capabilities. Beyond CPU use 
+cases, there are GPU accelerators with their own memory on board.
+The opportunities provided by modern heterogeneous memory platforms come 
+together with additional challenges. This means that additional software 
+changes might be required to fully leverage new HW capabilities. The are two 
+main problems that modern applications need to deal with. The first one is 
+appropriate data placement and data migration between different types of 
+memory. The second one is how SW should deal with different memory topologies.
+All applications can be divided into two big groups: enlightened and 
+unenlightened. Enlightened applications explicitly manage data allocation 
+distribution among memory tiers and further data migration. Unenlightened 
+applications do not require any code modifications and rely on underlying 
+infrastructure which is in turn enlightened. And underlying infrastructure is 
+not only OS with various memory tiering solutions to migrate memory pages 
+between tiers, but also middleware: frameworks and libraries. 
+
+==============
+ Architecture
+==============
+
+The Unified Memory Framework (UMF) is a library for constructing allocators 
+and memory pools. It also contains broadly useful abstractions and utilities 
+for memory management. UMF allows users to manage multiple memory pools 
+characterized by different attributes, allowing certain allocation types to be 
+isolated from others and allocated using different hardware resources as 
+required. 
+
+A memory pool is a combination of a pool allocator and one or more memory 
+targets accessed by memory providers along with their properties and allocation 
+policies. Specifically, a memory provider is responsible for coarse-grained 
+memory allocations, while the pool allocator controls the pool and handles 
+fine-grained memory allocations. UMF provides distinct interfaces for both pool 
+allocators and memory providers, allowing integration into various 
+applications. 
+
+.. figure:: ../assets/images/intro_architecture.png
+
+The UMF library contains various pool allocators and memory providers  but also 
+allows for the integration of external ones, giving users the flexibility to 
+either use existing solutions or provide their implementations. 
```

Comment:
`The UMF library contains a collection of predefined pool allocators and memory providers adhering to UMF interfaces. Users have the flexibility to interchange between predefined and user-created solutions.`

Reply:
I want to emphasize that the user can create their own providers and pools. This is one of the main features of UMF. Please give me suggestions on how I can do this better.

Reply:
What do you think of this?: `The UMF library contains a collection of predefined pool allocators and memory providers adhering to UMF interfaces. Users can opt for predefined solutions or use the same UMF interfaces to provide their own custom implementations.`

Reply:
thanks, done